README.md

```diff
@@ -15,251 +18,329 @@ The role can be used to configure:
+    to autoconnect another connection (for more details, see [rh#1401515](https://bugzilla.redhat.com/show_bug.cgi?id=1401515)).
+
+- `initscripts` deletes the ifcfg file in most cases with no impact on the system unless a component relies on the sysconfig directory.
+
+**Note**: For profiles that only contain a `state` option, the `network` role only activates
+or deactivates the connection without changing its configuration.
+
+
+### `type`
+
+The `type` option can be set to the following values:
 
-  - `bond`
-  - `bridge`
   - `ethernet`
-  - `infiniband`
-  - `macvlan`
+  - `bridge`
+  - `bond`
   - `team`
   - `vlan`
+  - `macvlan`
+  - `infiniband`
+
+#### `type: ethernet`
+
+If the type is `ethernet`, then there can be an extra `ethernet` dictionary with the following
+items (options): `autoneg`, `speed` and `duplex`, which correspond to the
+settings of the `ethtool` utility with the same name.
+
+* `autoneg`: `yes` (default) or `no` [if auto-negotiation is enabled or disabled]
+* `speed`: speed in Mbit/s
+* `duplex`: `half` or `full`
+
+Note that the `speed` and `duplex` link settings are required when autonegotiation is disabled (autoneg:no).
+
+#### `type: bridge`, `type: bond`, `type: team`
 
-The value `present` for the `persistent_state` setting does not directly
-result in a change in the network configuration. That is, without `state`
-set to `up`, the profile is only created or modified, not activated.
+The `bridge`, `bond`, `team` device types work similar. Note that `team` is not supported in RHEL6 kernels.
+
+For slaves, the `slave_type` and `master` properties must be set. Note that slaves should not have `ip` settings.
+
+The `master` refers to the `name` of a profile in the ansible
+Playbook. It is neither an interface-name nor a connection-id of
+NetworkManager.
+
+- For `NetworkManager`, `master` corresponds to the `connection.uuid`
+  of the corresponding profile.
+
+- For `initscripts`, `master` determines the `DEVICE` from the corresponding
+  ifcfg file.
+
+As `master` refers to other profiles of the same or another play,
+the order of the `connections` list matters. Also, `--check` ignores
+the value of the `master` and assumes it will be present during a real
+run. That means, in presence of an invalid `master`, `--check` may
+signal success but the actual play run fails.
+
+#### `type: vlan`
+
+Similar to `master`, the `parent` references the connection profile in the ansible
+role.
+
+#### `type: macvlan`
+
+Similar to `master` and `vlan`, the `parent` references the connection profile in the ansible
+role.
 
-- For NetworkManager, note the new connection profile is created with
-  `autoconnect` turned on by default. Thus, NetworkManager may very well decide
-  right away to activate the new profile on a currently disconnected device.
-  ([rh#1401515](https://bugzilla.redhat.com/show_bug.cgi?id=1401515)).
 
 ### `autoconnect`
 
+For NetworkManager, the new connection profile is created when the `autoconnect`
```

Comment:
This is wrong. The new profile is not created when the `autoconnect` option is enabled. It is created with the `autoconnect` property enabled.